eth/downloader/downloader.go

```diff
@@ -741,9 +752,36 @@ func (d *Downloader) findAncestor(p *peerConnection, remoteHeader *types.Header)
 		}
 	}
 
+	// Only attempt span search if local head is "close" to reported remote
+	if remoteHeight-localHeight <= uint64(MaxHeaderFetch) {
```

Comment:
Are we, at this point, certain that the remoteHeight is higher than localHeight? What if it's a lower height but higher TD? Shouldn't this check use `abs` ?

Reply:
Yeap, I think you're correct. Fixed with https://github.com/ethereum/go-ethereum/pull/21063/commits/58dd36b66f6ab27e93527869a417e8edaab3dd31